### DISCUSSION TITLE
Best way to use lazyload for the Facebook comments plugin?

### DISCUSSION THREAD

**Discussion by blogi-kriptovalyut** [mmistakes/minimal-mistakes#2669] [Q&A]
I have implemented lazy load for images. Now I would like to do the same for the facebook comments plugin.
What is the best way to do this?

Page-speed also one of the major factor in SEO!

Facebook widget lowers PageSpeed Insights from mobile to 18!

**Comment by mmistakes** [comment]
Google is your friend here: https://www.h3xed.com/web-development/how-to-lazy-load-the-facebook-comments-plugin-using-javascript

**Comment by blogi-kriptovalyut** [comment]
I've tried this option. It is 2014 and is probably out of date. I couldn't get it to run on Jekyll

**Comment by mmistakes** [comment]
Personally I don't use Facebook comments so I won't be any help. I try to avoid those 3rd party scripts (FB, Disqus, etc.) as you've noted they hurt page speed, not to mention they have some privacy concerns.

The theme just includes them to offer choice. For my sites I include comments statically, but that option isn't for everyone and has its own set of drawbacks.

**Comment by blogi-kriptovalyut** [comment]
I originally wanted to install staticman. But I was immediately stuck on the first step.
This username does not exist.

![img-2020-08-25-22-19-23](https://user-images.githubusercontent.com/50487083/91218035-25108c80-e721-11ea-9397-cec5e0c7cbab.png)

so i cant add Collaborators like it says here: https://staticman.net/docs/getting-started

**Comment by mmistakes** [comment]
Their docs are outdated. If you search through the issues in their repo there are a few threads that have links to various sites and blogs explaining how to setup your own free Heroku instance of Staticman.

That's really the only reliably way to use it. If you try and use the public `staticmanapp` instance you'll find comments don't make it through due to usage limits.

**Comment by mmistakes** [comment]
Also search around for lazy loading Disqus. I've seen plenty of solutions that either load the comment script when scrolling to it, or have a "Load comments" button that then loads the script. You can do the exact same thing with Facebook, just load their script instead.

There should be no limitation with Jekyll as it's just HTML and JavaScript. Jekyll could care less what you put on the page.

**Comment by blogi-kriptovalyut** [comment]
> Their docs are outdated. If you search through the issues in their repo there are a few threads that have links to various sites and blogs explaining how to setup your own free Heroku instance of Staticman.
> 
> That's really the only reliably way to use it. If you try and use the public `staticmanapp` instance you'll find comments don't make it through due to usage limits.

oh...this will take a long time.....https://travisdowns.github.io/blog/2020/02/05/now-with-comments.html

**Reply by VincentTam** [reply]
Hi this is the author of Staticman v3's documentation (<span>#</span>1956 & <span>#</span>2043).  I'm sorry for my inaction over the past year.  I've recently tested Staticman v3's GitHub App authorisation method (requested in eduardoboucas/staticman<span>#</span>243, implemented since eduardoboucas/staticman<span>#</span>255).  AFAIK, it's much better than the legacy Git* bot (described in the linked blog article) for the following reasons:

1. avoid API rate limit (eduardoboucas/staticman<span>#</span>222, eduardoboucas/staticman<span>#</span>227)
2. better compliance with the maintainer's recommendation to use a self-hosted instance (in the comments in eduardoboucas/staticman<span>#</span>317) as
    + one can make his/her GitHub App private and limit the GitHub repo(s) that the GitHub App can be installed to
    + the URL of your API instance can be found out using any modern browser's web dev tool.  Unless you separate your comments into a private repo, the user name of your Git* bot would be visible to all.  Anyone can invite your Git* bot.
